src/Nethermind/Nethermind.Evm/CodeAnalysis/JumpDestinationAnalyzer.cs

```diff
@@ -164,8 +213,34 @@ private static void MarkJumpDestinations(long[] jumpDestinationBitmap, int pos,
 
         public void Execute()
         {
-            // This is to support background thread preparation of the bitmap.
-            _jumpDestinationBitmap ??= CreateJumpDestinationBitmap(MachineCode.Span);
+            if (_jumpDestinationBitmap is null && Volatile.Read(ref _analysisComplete) is null)
+            {
+                ManualResetEventSlim analysisComplete = new(initialState: false);
+                if (Interlocked.CompareExchange(ref _analysisComplete, analysisComplete, null) is null)
+                {
+                    Thread thread = Thread.CurrentThread;
+                    ThreadPriority priority = thread.Priority;
+                    try
+                    {
+                        // Boost the priority of the thread as block processing may be waiting on this.
+                        thread.Priority = ThreadPriority.AboveNormal;
+
+                        _jumpDestinationBitmap ??= CreateJumpDestinationBitmap();
+
+                        // Release the MRES to be GC'd
+                        _analysisComplete = _jumpDestinationBitmap;
```

Comment:
Dispose MRES?

Reply:
MRES doesn't have a finalizer and is only when you access the WaitHandle property (as we do with `WaitHandleExtensions`) that it creates a finalizable object that needs to be disposed. 

Regular MRE needs to be disposed; but MRES doesn't use any OS sync primitives unless you access the WaitHandle and cause it to allocate one.